### PR TITLE
Add information about auth tokens

### PR DESCRIPTION
**Pull request by wtrocki**
for backend authentication.

Related change in code: https://github.com/aerogear/aerogear-android-sdk/pull/283/files

### CODE REVIEW

File: not-published/android/using-auth-sdk.adoc [hunk 1]
```diff
@@ -337,5 +337,16 @@ You can use the `AuthService#deleteTokens` function to delete the local authenti
 
 *Note:* To perform backchannel or federated logouts, you must enable the Backchannel Logout option for the federated identity provider. More information is available in the Keycloak documentation under  http://www.keycloak.org/docs/latest/server_admin/index.html#openid-connect-v1-0-identity-providers[OIDC Identity Providers].
```
Comment (finp) [inline not-published/android/using-auth-sdk.adoc]: @pmacko1 could you edit this please?

Comment (pmacko1) [reply not-published/android/using-auth-sdk.adoc]: yep, done

**Comment by wtrocki** [comment]
I need tech aproval: @ziccardi @danielpassos

**Comment by cfoskin** [comment]
@wtrocki I think @danielpassos  might be better placed than me to provide tech approval here

File: not-published/android/using-auth-sdk.adoc (hunk 1 above)
Comment (finp) [reply not-published/android/using-auth-sdk.adoc]: @wtrocki Note that this doc is not actually published, ie it got moved to this repo, but is not in the published struture :(

Comment (wtrocki) [reply not-published/android/using-auth-sdk.adoc]: @finp so where we should add this then?

File: not-published/android/using-auth-sdk.adoc [hunk 2]
```diff
@@ -335,7 +335,18 @@ authService.logout(currentUser, new Callback<UserPrincipal>() {
...
+=== Using auth with backend
+
+To authenticate service on existing backend server users need to supply additional token:
+
+[source]
+----
+Authorization: BEARER Yourtoken
+----
+
+Auth SDK also provides _AuthHeaderProvider_ class that can be used for this purpose.
```
Comment (danielpassos) [inline not-published/android/using-auth-sdk.adoc]: Maybe we should add some code example?

```java
MobileCore.getInstance().getHttpLayer().requestHeaderInterceptor().add(yourAuthHeaderProviderInstance);
```

NOR

We say we have a util method on authService to help you on it

```java
﻿MobileCore.getInstance().getHttpLayer().requestHeaderInterceptor().add(authService.getAuthHeaderProvider());
```

**Comment by wtrocki** [comment]
Closing as I put changes in wrong place.

**Comment by wtrocki** [comment]
We needed to add this to IOS platform first in order to add it to generic docs. I believe that this is now ready to be added. ping @ziccardi

**Comment by ziccardi** [comment]
@wtrocki, @laurafitzgerald  both feature and docs have been added for ios and xamarin. Waiting for a final approve from @finp, then it will be merged.

**Comment by finp** [comment]
@ziccardi  I assume you're referring to https://github.com/aerogear/mobile-docs/pull/484 ?

**Comment by ziccardi** [comment]
@finp : yep!

**Pull request closed without merging**